src/completion.ts

```diff
@@ -108,14 +109,21 @@ export function asCompletionItem(entry: tsp.CompletionEntry, file: string, posit
         if (!insertText) {
             insertText = item.label;
         }
-
-        item.textEdit = lsp.TextEdit.replace(replacementRange, insertText);
+        item.textEdit = createTextEdit(replacementRange, position, insertText, features);
     } else {
         item.insertText = insertText;
     }
     return item;
 }
 
+function createTextEdit(replacementRange: lsp.Range, position: lsp.Position, insertText: string, features: SupportedFeatures): lsp.TextEdit | lsp.InsertReplaceEdit {
+    if (features.completionInsertReplaceSupport) {
+        const insertRange = lsp.Range.create(replacementRange.start, position);
+        return lsp.InsertReplaceEdit.create(insertText, insertRange, replacementRange);
+    }
+    return lsp.TextEdit.replace(replacementRange, insertText);
+}
```

Comment:
As I've alluded earlier, I don't think this function adds value and it just adds extra unnecessary lines.
We can inline this code.